apps/patient-listener/src/main/java/gov/cdc/nbs/patientlistener/service/PatientCreator.java

```diff
@@ -0,0 +1,125 @@
+package gov.cdc.nbs.patientlistener.service;
+
+import gov.cdc.nbs.address.City;
+import gov.cdc.nbs.address.Country;
+import gov.cdc.nbs.address.County;
+import gov.cdc.nbs.entity.odse.Person;
+import gov.cdc.nbs.message.PatientCreateRequest;
+import gov.cdc.nbs.patient.PatientCommand;
+import gov.cdc.nbs.repository.PersonRepository;
+import org.springframework.stereotype.Component;
+
+import javax.transaction.Transactional;
+
+@Component
+class PatientCreator {
+
+    private final PersonRepository repository;
+
+    PatientCreator(final PersonRepository repository) {
+        this.repository = repository;
+    }
+
+
+    @Transactional
+    Person create(final PatientCreateRequest request) {
+
+        Person person = new Person(asAdd(request));
+
+        request.names().stream().map(name -> as(request, name))
+                .forEach(person::add);
+
+        request.races().stream().map(race -> asRace(request, race))
+                .forEach(person::add);
+
+        request.addresses().stream().map(address -> as(request, address))
+                .forEach(person::add);
+
+        request.phoneNumbers().stream().map(phoneNumber -> as(request, phoneNumber))
+                .forEach(person::add);
+
+        request.emailAddresses().stream().map(emailAddress -> asEmailAddress(request, emailAddress))
+                .forEach(person::add);
+
+
+        return repository.save(person);
+    }
+
+    private PatientCommand.AddPatient asAdd(final PatientCreateRequest request) {
+        return new PatientCommand.AddPatient(
+                request.patient(),
+                request.patientLocalId(),
+                request.ssn(),
+                request.dateOfBirth(),
+                request.birthGender(),
+                request.currentGender(),
+                request.deceased(),
+                request.deceasedTime(),
+                request.maritalStatus(),
+                request.ethnicity(),
+                request.createdBy(),
+                request.createdAt()
+        );
+    }
+
+    private PatientCommand.AddName as(final PatientCreateRequest request, final PatientCreateRequest.Name name) {
```

Comment:
can we rename the `as` methods to `asName`, `asPostalAddresses`, etc to match the `PatientCreateRequestResolver` naming

Reply:
Sure can